federation/pkg/kubefed/init/init_test.go

```diff
@@ -1444,7 +1444,8 @@ func tlsHandshake(t *testing.T, sCfg, cCfg *tls.Config) error {
 		}
 	}()
 
-	c, err := tls.Dial("tcp", s.Addr().String(), cCfg)
+	addr := strings.Split(s.Addr().String(), ":")
```

Comment:
I looked at the issue filed. It looks like you somehow have network protocol default to IPv6 in your environment. Shouldn't you make that default to IPv4 when running these tests instead of this hacky solution in the code base?

Even if you were to make a change here, the right way to do this is to force the address to IPv4 by passing `tcp4` instead of `tcp` to `tls.Listen()` call above.